engine/src/flutter/shell/platform/android/io/flutter/embedding/engine/systemchannels/SensitiveContentChannel.java

```diff
@@ -0,0 +1,160 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine.systemchannels;
+
+import android.view.View;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+import io.flutter.Log;
+import io.flutter.embedding.engine.dart.DartExecutor;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.StandardMethodCodec;
+
+/**
+ * {@link SensitiveContentChannel} is a platform channel that is used by the framework to get and
+ * set the content sensitivity of a native Flutter Android {@code View}.
+ */
+public class SensitiveContentChannel {
+  private static final String TAG = "SensitiveContentChannel";
+
+  /**
+   * Flutter ContentSensitivity.autoSensitive name that represents Android's
+   * `View.CONTENT_SENSITIVITY_AUTO` setting.
+   *
+   * <p>See https://developer.android.com/reference/android/view/View#CONTENT_SENSITIVITY_AUTO.
+   */
+  @VisibleForTesting
+  public static final String AUTO_SENSITIVE_CONTENT_SENSITIVITY = "autoSensitive";
+
+  /**
+   * Flutter ContentSensitivity.sensitive name that represents Android's
+   * `View.CONTENT_SENSITIVITY_SENSITIVE` setting.
+   *
+   * <p>See https://developer.android.com/reference/android/view/View#CONTENT_SENSITIVITY_SENSITIVE.
+   */
+  @VisibleForTesting public static final String SENSITIVE_CONTENT_SENSITIVITY = "sensitive";
+
+  /**
+   * Flutter ContentSensitivity.notSensitive name that represents Android's
+   * `View.CONTENT_SENSITIVITY_NOT_SENSITIVE` setting.
+   *
+   * <p>See
+   * https://developer.android.com/reference/android/view/View#CONTENT_SENSITIVITY_NOT_SENSITIVE.
+   */
+  @VisibleForTesting public static final String NOT_SENSITIVE_CONTENT_SENSITIVITY = "notSensitive";
+
+  public final MethodChannel channel;
+  private SensitiveContentMethodHandler sensitiveContentMethodHandler;
+
+  @NonNull
+  public final MethodChannel.MethodCallHandler parsingMethodHandler =
+      new MethodChannel.MethodCallHandler() {
+        @Override
+        public void onMethodCall(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
+          if (sensitiveContentMethodHandler == null) {
+            // No SensitiveContentChannel registered, call not forwarded to sensitive content
+            // API.");
+            return;
+          }
+          String method = call.method;
+          Log.v(TAG, "Received '" + method + "' message.");
+          switch (method) {
```

Comment:
Note to reviewers: I refactored this handler to only catch exceptions I throw in `SensitiveContentPlugin` to signal various errors that could be made by the user.